test/framework/kubetest/run.go

```diff
@@ -143,27 +137,51 @@ func Run(ctx context.Context, input RunInput) error {
 	if input.ConformanceImage == "" {
 		input.ConformanceImage = versionToConformanceImage(input.KubernetesVersion)
 	}
-	kubeConfigVolumeMount := volumeArg(tmpKubeConfigPath, "/tmp/kubeconfig")
-	outputVolumeMount := volumeArg(reportDir, "/output")
-	viperVolumeMount := volumeArg(tmpConfigFilePath, "/tmp/viper-config.yaml")
+	volumeMounts := map[string]string{
+		tmpKubeConfigPath: "/tmp/kubeconfig",
+		reportDir:         "/output",
+		tmpConfigFilePath: "/tmp/viper-config.yaml",
+	}
 	user, err := user.Current()
 	if err != nil {
 		return errors.Wrap(err, "unable to determine current user")
 	}
-	userArg := user.Uid + ":" + user.Gid
-	networkArg := "--network=kind"
-	e2eCmd := exec.Command("docker", "run", "--user", userArg, kubeConfigVolumeMount, outputVolumeMount, viperVolumeMount, "-t", networkArg)
-	if len(testRepoListVolumeArgs) > 0 {
-		e2eCmd.Args = append(e2eCmd.Args, testRepoListVolumeArgs...)
+	env := map[string]string{}
+
+	if input.KubeTestRepoListPath != "" {
+		tmpKubeTestRepoListPath := path.Join(kubetestConfigDir, "repo_list.yaml")
+		if err := copyFile(input.KubeTestRepoListPath, tmpKubeTestRepoListPath); err != nil {
+			return err
+		}
+		dest := "/tmp/repo_list.yaml"
+		env["KUBE_TEST_REPO_LIST"] = dest
+		volumeMounts[tmpKubeTestRepoListPath] = dest
 	}
-	e2eCmd.Args = append(e2eCmd.Args, input.ConformanceImage)
-	e2eCmd.Args = append(e2eCmd.Args, "/usr/local/bin/ginkgo")
-	e2eCmd.Args = append(e2eCmd.Args, ginkgoArgs...)
-	e2eCmd.Args = append(e2eCmd.Args, "/usr/local/bin/e2e.test")
-	e2eCmd.Args = append(e2eCmd.Args, "--")
-	e2eCmd.Args = append(e2eCmd.Args, e2eArgs...)
-	e2eCmd = framework.CompleteCommand(e2eCmd, "Running e2e test", false)
-	if err := e2eCmd.Run(); err != nil {
+
+	// Formulate our command arguments
+	args := []string{"/usr/local/bin/ginkgo"}
+	args = append(args, ginkgoArgs...)
+	args = append(args, "/usr/local/bin/e2e.test")
+	args = append(args, "--")
+	args = append(args, e2eArgs...)
+
+	// Get our current working directory. Just for information, so we don't need
+	// to worry about errors at this point.
+	cwd, _ := os.Getwd()
+	Byf("Running e2e test: dir=%s, command=%q", cwd, args)
+
+	err = dockerexec.RunContainer(
```

Comment:
I think the "exec" package name is a bit confusing. Because it's not actually an "os exec" anymore. Maybe we should rename it?

Reply:
Yeah, that's a good point. I'm not sure what would be the best naming. Maybe we can refactor this in a follow up? The `exec` package does have some exec calls for kubectl and general commands, so either we should move the docker things out of their into another package, or we should rename `exec` into something more general.

Though I guess technically it is still executing some operations outside the current program - just not by what someone may assume would be a command execution.

Reply:
We can maybe put the docker.go in a separate `container` package or something similar. I assume it will "grow" in this direction in the next PRs. But let's see what @fabriziopandini thinks about it

Reply:
I don't think this is the end state.
There is lot of overlap between the code in this file and what we have to do into CAPD, so I do expect us to get our CRI wrapper implemented somewhere, might be in CAPD, might be in a separated subproject under /test (given that we will be required to share this between CAPD and the E2E framework)
However we are not yet there in this work, unless @stmcginnis would like to tackle this point now given the go.mod issues raised by @vince

Reply:
I could create a `test/cri` subproject to move the runtime interaction code under. For CAPD, we are working towards putting the implementation details behind an abstraction interface for testing and (long term) potentially (maybe!) being able to swap container runtimes.

So I could start fleshing that out and start with what we need for the test framework. That would get these requirements out of the way for non-testing and non-CAPD cases. Then after this test framework merges, I can continue on with the changes we want to do for CAPD.